src/store/index.ts

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/*
+   This file is part of Astarte.
+
+   Copyright 2021 Ispirata Srl
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import { configureStore } from '@reduxjs/toolkit';
+import { useDispatch, useSelector, Selector } from 'react-redux';
+import type AstarteClient from 'astarte-client';
+import _ from 'lodash';
+
+import blocksSlice, { BlocksState } from './blocks';
+
+type StoreState = {
+  blocks: BlocksState;
+};
+
+const storeSlices = {
+  blocks: blocksSlice,
+};
+
+const actions = _.mapValues(storeSlices, (slice) => slice.actions);
+const selectors = _.mapValues(storeSlices, (slice) => slice.selectors);
+const reducer = _.mapValues(storeSlices, (slice) => slice.reducer);
+
+const createStore = (astarte: AstarteClient) =>
```

Comment:
This smells like premature optimization. Let's write a specific store first, and if in the future we will need a generic store, we can create something that fits all the specific stores requirements.